src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -1671,47 +1671,310 @@ static double NegativeN(double x, int n)
         /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.Acos(TSelf)" />
         public static double Acos(double x) => Math.Acos(x);
 
+        /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.AcosPi(TSelf)" />
+        public static double AcosPi(double x)
+        {
+            return Acos(x) / Pi;
+        }
+
         /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.Asin(TSelf)" />
         public static double Asin(double x) => Math.Asin(x);
 
+        /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.AsinPi(TSelf)" />
+        public static double AsinPi(double x)
+        {
+            return Asin(x) / Pi;
+        }
+
         /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.Atan(TSelf)" />
         public static double Atan(double x) => Math.Atan(x);
 
         /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.Atan2(TSelf, TSelf)" />
         public static double Atan2(double y, double x) => Math.Atan2(y, x);
 
+        /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.Atan2Pi(TSelf, TSelf)" />
+        public static double Atan2Pi(double y, double x)
+        {
+            return Atan2(y, x) / Pi;
+        }
+
+        /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.AtanPi(TSelf)" />
+        public static double AtanPi(double x)
+        {
+            return Atan(x) / Pi;
+        }
+
         /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.Cos(TSelf)" />
         public static double Cos(double x) => Math.Cos(x);
 
+        /// <inheritdoc cref="ITrigonometricFunctions{TSelf}.CosPi(TSelf)" />
+        public static double CosPi(double x)
+        {
+            // This code is based on `cospi` from amd/aocl-libm-ose
+            // Copyright (C) 2008-2020 Advanced Micro Devices, Inc. All rights reserved.
+            //
+            // Licensed under the BSD 3-Clause "New" or "Revised" License
+            // See THIRD-PARTY-NOTICES.TXT for the full license text
+
+            double result;
+
+            if (IsFinite(x))
```

Comment:
This is done for better perf. It being `NaN` is "unlikely" and we don't have a way to tell the JIT that. By default (barring PGO data indicating otherwise), its going to be `if (cond) { predicted } unpredicted` and so this helps with the CPU speculative execution doing the right thing.